src/main/java/de/dennisguse/opentracks/services/TrackRecordingService.java

```diff
@@ -179,17 +181,29 @@ public void onDestroy() {
 
         PreferencesUtils.unregister(this, sharedPreferenceChangeListener);
 
+
         try {
             voiceExecutor.shutdown();
         } finally {
             voiceExecutor = null;
         }
 
+        if (PreferencesUtils.shouldInstantExportAfterWorkout(getApplicationContext())) {
+            SharedPreferences sharedPreferences = PreferencesUtils.getSharedPreferences(this);
+            TrackFileFormat trackFileFormat = PreferencesUtils.getExportTrackFileFormat(this);
+            InstantPostWorkoutExport instantPostWorkoutExport = new InstantPostWorkoutExport(getApplicationContext(), contentProviderUtils, trackFileFormat);
+
+            String singleExportDirectorySettingsKey = getString(R.string.settings_single_export_directory_key);
+            String directoryAsString = sharedPreferences.getString(singleExportDirectorySettingsKey, "invalid uri");
```

Comment:
i did
```
public static Uri getInstantExportDirectoryUri(Context context)
    {
        String singleExportDirectorySettingsKey = getString(context, R.string.settings_single_export_directory_key, null);
        String directoryAsString = getSharedPreferences(context).getString(singleExportDirectorySettingsKey, "invalid uri");
        return Uri.parse(directoryAsString);
    }
```
but that gives me only `invalid url`. With the previous implementation it worked

Reply:
Then the key is not set and you get the default value.
Can you check that you are setting the value for this key successfully?
Because it looks good.

Reply:
the second line of code is totally useless (copy and paste error). it works now

Reply:
done